### ISSUE TITLE
Function definition with macro

### ISSUE THREAD

**Issue by wangl-cc** [fonsp/Pluto.jl#1236]
If I defined a  macro that converts a function definition expression to a non-function definition expression, such as `@macroexpand`, then
a cell
```julia
@macroexpand f(x) = x
```
and a cell
```julia
f(x) = x
```
will throw a multiple definitions error. Even the first expression isn't actually a function definition.

However, if I defined a  macro that converts a non-function definition expression to a function definition expression, like
```julia
macro genfunc(name)
    name = esc(name)
    :($name() = 1)
end
```
then multiple same cells
```julia
@genfunc f
```
will not throw any error message.

I know `Pluto` doesn't understand macros now and #1032 may fix it.
These may be a test of it?

**Comment by Pangoraw** [comment]
Hi! Indeed, this should be partly fixed by #1032. You can try it out by running:
```julia
julia> ]
pkg> activate --temp
pkg> add https://github.com/Pangoraw/Pluto.jl#macros
julia> import Pluto; Pluto.run()
```
You may have to re-run the cell calling the macro after running the cell that defines the macro. Tell me if you encounter any problems!

**Comment by wangl-cc** [comment]
Yes, it is fixed on your branch. Thanks a lot.
